# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1249372538 -7200
#      Tue Aug 04 09:55:38 2009 +0200
# Branch release-3-2-x
# Node ID 3df527f71cee4c32e44203a52190b268f45138eb
# Parent  bf90777b58c95f5a794e2a458d6e44f1f9ec7bd3
use complex function for acos mapper if arg is out of range [-1, 1]

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,14 @@
+2009-07-29  John W. Eaton  <jwe@octave.org>
+
+	* ov-float.cc, ov-flt-re-mat.cc, ov-re-mat.cc, ov-re-sparse.cc,
+	ov-scalar.cc: Use complex function for acos mapper if arg is out
+	of range [-1, 1].
+
 2009-07-29  John W. Eaton  <jwe@octave.org>
 
 	* parse.y (param_list_end): Also set
 	lexer_flags.looking_for_object_index to false.
 
 2009-07-24  John W. Eaton  <jwe@octave.org>
 
 	* pt-mat.cc (DO_SINGLE_TYPE_CONCAT_NO_MUTATE): New macro.
diff --git a/src/ov-float.cc b/src/ov-float.cc
--- a/src/ov-float.cc
+++ b/src/ov-float.cc
@@ -274,17 +274,17 @@ xconj (float x)
   return x;
 }
 
 SCALAR_MAPPER (erf, ::erff)
 SCALAR_MAPPER (erfc, ::erfcf)
 SCALAR_MAPPER (gamma, xgamma)
 CD_SCALAR_MAPPER (lgamma, xlgamma, xlgamma, 0.0, octave_Float_Inf)
 SCALAR_MAPPER (abs, ::fabsf)
-SCALAR_MAPPER (acos, ::acosf)
+CD_SCALAR_MAPPER (acos, ::acosf, ::acos, -1.0, 1.0)
 CD_SCALAR_MAPPER (acosh, ::acoshf, ::acosh, 1.0, octave_Float_Inf)
 SCALAR_MAPPER (angle, ::arg)
 SCALAR_MAPPER (arg, ::arg)
 CD_SCALAR_MAPPER (asin, ::asinf, ::asin, -1.0, 1.0)
 SCALAR_MAPPER (asinh, ::asinhf)
 SCALAR_MAPPER (atan, ::atanf)
 CD_SCALAR_MAPPER (atanh, ::atanhf, ::atanh, -1.0, 1.0)
 SCALAR_MAPPER (ceil, ::ceilf)
diff --git a/src/ov-flt-re-mat.cc b/src/ov-flt-re-mat.cc
--- a/src/ov-flt-re-mat.cc
+++ b/src/ov-flt-re-mat.cc
@@ -778,17 +778,17 @@ octave_float_matrix::finite (void) const
 {
   return matrix.isfinite ();
 }
 
 ARRAY_MAPPER (erf, FloatNDArray::dmapper, ::erff)
 ARRAY_MAPPER (erfc, FloatNDArray::dmapper, ::erfcf)
 ARRAY_MAPPER (gamma, FloatNDArray::dmapper, xgamma)
 CD_ARRAY_MAPPER (lgamma, xlgamma, xlgamma, 0.0, octave_Float_Inf)
-ARRAY_MAPPER (acos, FloatNDArray::dmapper, ::acosf)
+CD_ARRAY_MAPPER (acos, ::acosf, ::acos, -1.0, 1.0)
 CD_ARRAY_MAPPER (acosh, ::acoshf, ::acosh, 1.0, octave_Float_Inf)
 ARRAY_MAPPER (angle, FloatNDArray::dmapper, ::arg)
 ARRAY_MAPPER (arg, FloatNDArray::dmapper, ::arg)
 CD_ARRAY_MAPPER (asin, ::asinf, ::asin, -1.0, 1.0)
 ARRAY_MAPPER (asinh, FloatNDArray::dmapper,::asinhf)
 ARRAY_MAPPER (atan, FloatNDArray::dmapper, ::atanf)
 CD_ARRAY_MAPPER (atanh, ::atanhf, ::atanh, -1.0, 1.0)
 ARRAY_MAPPER (ceil, FloatNDArray::dmapper, ::ceilf)
diff --git a/src/ov-re-mat.cc b/src/ov-re-mat.cc
--- a/src/ov-re-mat.cc
+++ b/src/ov-re-mat.cc
@@ -806,17 +806,17 @@ octave_matrix::finite (void) const
 {
   return matrix.isfinite ();
 }
 
 ARRAY_MAPPER (erf, NDArray::dmapper, ::erf)
 ARRAY_MAPPER (erfc, NDArray::dmapper, ::erfc)
 ARRAY_MAPPER (gamma, NDArray::dmapper, xgamma)
 CD_ARRAY_MAPPER (lgamma, xlgamma, xlgamma, 0.0, octave_Inf)
-ARRAY_MAPPER (acos, NDArray::dmapper, ::acos)
+CD_ARRAY_MAPPER (acos, ::acos, ::acos, -1.0, 1.0)
 CD_ARRAY_MAPPER (acosh, ::acosh, ::acosh, 1.0, octave_Inf)
 ARRAY_MAPPER (angle, NDArray::dmapper, ::arg)
 ARRAY_MAPPER (arg, NDArray::dmapper, ::arg)
 CD_ARRAY_MAPPER (asin, ::asin, ::asin, -1.0, 1.0)
 ARRAY_MAPPER (asinh, NDArray::dmapper,::asinh)
 ARRAY_MAPPER (atan, NDArray::dmapper, ::atan)
 CD_ARRAY_MAPPER (atanh, ::atanh, ::atanh, -1.0, 1.0)
 ARRAY_MAPPER (ceil, NDArray::dmapper, ::ceil)
diff --git a/src/ov-re-sparse.cc b/src/ov-re-sparse.cc
--- a/src/ov-re-sparse.cc
+++ b/src/ov-re-sparse.cc
@@ -871,17 +871,17 @@ xconj (double x)
   return x;
 }
 
 SPARSE_MAPPER (erf, SparseMatrix::dmapper, ::erf)
 SPARSE_MAPPER (erfc, SparseMatrix::dmapper, ::erfc)
 SPARSE_MAPPER (gamma, SparseMatrix::dmapper, xgamma)
 CD_SPARSE_MAPPER (lgamma, xlgamma, xlgamma, 0.0, octave_Inf)
 SPARSE_MAPPER (abs, SparseMatrix::dmapper, ::fabs)
-SPARSE_MAPPER (acos, SparseMatrix::dmapper, ::acos)
+CD_SPARSE_MAPPER (acos, ::acos, ::acos, -1.0, 1.0)
 CD_SPARSE_MAPPER (acosh, ::acosh, ::acosh, 1.0, octave_Inf)
 SPARSE_MAPPER (angle, SparseMatrix::dmapper, ::arg)
 SPARSE_MAPPER (arg, SparseMatrix::dmapper, ::arg)
 CD_SPARSE_MAPPER (asin, ::asin, ::asin, -1.0, 1.0)
 SPARSE_MAPPER (asinh, SparseMatrix::dmapper, ::asinh)
 SPARSE_MAPPER (atan, SparseMatrix::dmapper, ::atan)
 CD_SPARSE_MAPPER (atanh, ::atanh, ::atanh, -1.0, 1.0)
 SPARSE_MAPPER (ceil, SparseMatrix::dmapper, ::ceil)
diff --git a/src/ov-scalar.cc b/src/ov-scalar.cc
--- a/src/ov-scalar.cc
+++ b/src/ov-scalar.cc
@@ -289,17 +289,17 @@ xconj (double x)
   return x;
 }
 
 SCALAR_MAPPER (erf, ::erf)
 SCALAR_MAPPER (erfc, ::erfc)
 SCALAR_MAPPER (gamma, xgamma)
 CD_SCALAR_MAPPER (lgamma, xlgamma, xlgamma, 0.0, octave_Inf)
 SCALAR_MAPPER (abs, ::fabs)
-SCALAR_MAPPER (acos, ::acos)
+CD_SCALAR_MAPPER (acos, ::acos, ::acos, -1.0, 1.0)
 CD_SCALAR_MAPPER (acosh, ::acosh, ::acosh, 1.0, octave_Inf)
 SCALAR_MAPPER (angle, ::arg)
 SCALAR_MAPPER (arg, ::arg)
 CD_SCALAR_MAPPER (asin, ::asin, ::asin, -1.0, 1.0)
 SCALAR_MAPPER (asinh, ::asinh)
 SCALAR_MAPPER (atan, ::atan)
 CD_SCALAR_MAPPER (atanh, ::atanh, ::atanh, -1.0, 1.0)
 SCALAR_MAPPER (ceil, ::ceil)
